include/zephyr/net/net_if.h

```diff
@@ -193,12 +194,32 @@ enum net_if_flag {
 	/** Interface supports IPv6 */
 	NET_IF_IPV6,
 
+	/** Interface up and running (ready to receive and transmit). */
+	NET_IF_RUNNING,
+
+	/** Driver signals L1 is up. */
+	NET_IF_LOWER_UP,
+
+	/** Driver signals dormant. */
+	NET_IF_DORMANT,
+
 /** @cond INTERNAL_HIDDEN */
 	/* Total number of flags - must be at the end of the enum */
 	NET_IF_NUM_FLAGS
 /** @endcond */
 };
 
+/** Network interface operational status (RFC 2863). */
+enum net_if_oper_state {
+	NET_IF_OPER_UNKNOWN,
+	NET_IF_OPER_NOTPRESENT,
+	NET_IF_OPER_DOWN,
+	NET_IF_OPER_LOWERLAYERDOWN,
+	NET_IF_OPER_TESTING,
+	NET_IF_OPER_DORMANT,
+	NET_IF_OPER_UP,
+};
```

Comment:
If we set this to
```
enum net_if_oper_state {
....
} __packed;
```
then the `oper_state` variable below can be set to `enum net_if_oper_state oper_state` and it would consume only one byte.